### PR TITLE
Add more API documentation in user guide

### PR DESCRIPTION
**Pull request by rahulporuri**
API documentation for the following functions are added in this PR

- ``deeplabcut.analyze_videos``
- ``deeplabcut.analyzeskeleton``
- ``deeplabcut.create_labeled_video``
- ``deeplabcut.create_training_model_comparison``
- ``deeplabcut.evaluate_network``
- ``deeplabcut.extract_outlier_frames``
- ``deeplabcut.filterpredictions``
- ``deeplabcut.label_frames``
- ``deeplabcut.merge_datasets``
- ``deeplabcut.plot_trajectories``
- ``deeplabcut.refine_labels``

Note that additionally, this PR updates the GitHub Action that builds Jupyter Book documentation to install GUI dependencies. This is necessary as the functions - ``deeplabcut.label_frames`` and ``deeplabcut.refine_labels`` - live in modules which rely on GUI dependencies.

Finally, with the changes introduced in this PR, there are no links to the "DOCSTRINGS" wiki page in the user guide.

### CODE REVIEW

**Comment by rahulporuri** [comment]
To-do, potentially in a later PR

- [ ] `deeplabcut.label_frames`
- [ ] `deeplabcut.refine_labels`

**Comment by rahulporuri** [comment]
@AlexEMG and @MMathisLab - This PR completes the first part of the API documentation effort. It's much bigger than I initially anticipated. I can split it up into multiple PRs e.g. one per function if that will speed up the review process.

**Comment by MMathisLab** [comment]
Looks good to me - thanks again @rahulporuri -- something I wanted to get around to for a long time, and really appreciate your contributions 🥇

File: deeplabcut/pose_estimation_tensorflow/predict_videos.py [hunk 1]
```diff
@@ -282,128 +282,187 @@ def analyze_videos(
...
+        be exported in comma-separated values format (.csv), which in turn can be
+        imported in many programs, such as MATLAB, R, Prism, etc.
 
     Examples
     --------
 
-    Windows example for analyzing 1 video
-    >>> deeplabcut.analyze_videos('C:\\myproject\\reaching-task\\config.yaml',['C:\\yourusername\\rig-95\\Videos\\reachingvideo1.avi'])
-    --------
+    Analysing a single video on Windows
```
Comment (jeylau) [inline deeplabcut/pose_estimation_tensorflow/predict_videos.py]:
```suggestion
    Analyzing a single video on Windows
```

File: deeplabcut/post_processing/filtering.py [hunk 2]
```diff
@@ -81,75 +81,113 @@ def filterpredictions(
...
+            'C:\\myproject\\reaching-task\\config.yaml',
+            ['C:\\myproject\\trailtracking-task\\test.mp4'],
+            shuffle=3,
+            filterype='arima',
+            ARdegree=5,
+            MAdegree=2,
+        )
 
-    One can then use the filtered rather than the frame-by-frame predictions by calling:
+    Use median filter over 10bins:
```
Comment (jeylau) [inline deeplabcut/post_processing/filtering.py]:
```suggestion
    Use median filter over 10 bins:
```

File: deeplabcut/utils/plotting.py [hunk 3]
```diff
@@ -177,61 +177,82 @@ def plot_trajectories(
...
+    videotype: str, optional, default=""
+        Checks for the extension of the video in case the input to the video is a
+        directory. Only videos with this extension are analyzed.
+        If left unspecified, videos with common extensions
+        ('avi', 'mp4', 'mov', 'mpeg', 'mkv') are kept.
 
-    shuffle: list, optional
-    List of integers specifying the shuffle indices of the training dataset. The default is [1]
+    shuffle: list, optional, default=1
+        List of integers specifying the shuffle indices of the training dataset.
```
Comment (jeylau) [inline deeplabcut/utils/plotting.py]: Just noticed that the doc was originally wrong here. It should read:
```suggestion
    shuffle: int, optional, default=1
        Integer specifying the shuffle index of the training dataset.
```

**Comment by jeylau** [comment]
I just left some very minor comments @rahulporuri, but thank you again for the contribution! The docs really look great! 🥰

**Comment by rahulporuri** [comment]
@jeylau thanks for the suggestions - I have incorporated all of them.

@MMathisLab thank you (and your lab/team) for putting together an amazing package and making it open source.

**Review by MMathisLab — APPROVED** [review]

**Pull request merged**